rasterio/rio/sample.py

```diff
@@ -0,0 +1,94 @@
+import json
+import logging
+import sys
+import warnings
+
+import click
+
+import rasterio
+from rasterio.rio.cli import cli
+
+
+warnings.simplefilter('default')
+
+
+@cli.command(short_help="Sample a dataset.")
+@click.argument('files', nargs=-1, required=True, metavar='FILE "[x, y]"')
+@click.option('--bidx', default=None, help="Indexes of input file bands.")
+@click.pass_context
+def sample(ctx, files, bidx):
+    """Sample a dataset at one or more points
+
+    Sampling points (x, y) encoded as JSON arrays, in the coordinate
+    reference system of the dataset, are read from the second
+    positional argument or stdin. Values of the dataset's bands
+    are also encoded as JSON arrays and are written to stdout.
+
+    Example:
+
+    \b
+        $ cat << EOF | rio sample tests/data/RGB.byte.tif
+        > [220650, 2719200]
+        > [219650, 2718200]
+        > EOF
+        [28, 29, 27]
+        [25, 29, 19]
+
+    By default, rio-sample will sample all bands. Optionally, bands
+    may be specified using a simple syntax:
+
+      --bidx N samples the Nth band (first band is 1).
+
+      --bidx M,N,0 samples bands M, N, and O.
+
+      --bidx M..O samples bands M-O, inclusive.
```

Comment:
I feel python slice syntax with 1-indexed bands would be extra confusing. What I'm going for with `--bidx M,N,0` and `--bidx M..O` is bash/ksh array syntax, as in

```
$ echo {3..5}
3 4 5
```

rio-sample is developed for the shell, shell-like syntax feels like a better fit to me.